client/python/openlineage/client/transport/msk_iam.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import functools
+import logging
+import os
+from typing import Any
+
+import attr
+from openlineage.client.transport import KafkaConfig, KafkaTransport
+
+log = logging.getLogger(__name__)
+
+
+def _detect_running_region() -> None | str:
+    """Dynamically determine the region from a running Glue job (or anything on EC2 for
+    that matter).
+    https://stackoverflow.com/questions/37514810/how-to-get-the-region-of-the-current-user-from-boto
+    """
+    import boto3  # type: ignore[import]
+
+    easy_checks = [
+        # check if set through ENV vars
+        os.environ.get("AWS_REGION"),
+        os.environ.get("AWS_DEFAULT_REGION"),
+        # else check if set in config or in boto already
+        boto3.DEFAULT_SESSION.region_name if boto3.DEFAULT_SESSION else None,
+        boto3.Session().region_name,
+    ]
```

Comment:
Is the order in which we search for the region documented somewhere?  F.e. if i set the region with both env variable and then in boto3.DEFAULT_SESSION. I did not see any test for that, so just asking out of curiosity. Maybe it's natural for somebody familiar with AWS, let me know 😄

Reply:
Not really, maybe is better just to do `boto3.Session().region_name` that should run all the checks that are there.

Reply:
The `AWS_DEFAULT_REGION ` can be remove. I test it better. 
But in general I could be better to fail if the region is not selected since the cluster could be in another one and it will just fail later by not sending any message make it more difficult to debug.